command/operator_debug_test.go

```diff
@@ -16,71 +16,37 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func Test_BadCSIPluginNames(t *testing.T) {
-	// Start test server and API client
-	srv, _, url := testServer(t, false, nil)
-	defer srv.Shutdown()
-
-	// Wait for leadership to establish
-	testutil.WaitForLeader(t, srv.Agent.RPC)
-
-	cases := []string{
-		"aws/ebs",
-		"gcp-*-1",
-	}
-	for _, pluginName := range cases {
-		cleanup := state.CreateTestCSIPlugin(srv.Agent.Server().State(), pluginName)
-		defer cleanup()
-	}
-
-	// Setup mock UI
-	ui := cli.NewMockUi()
-	cmd := &OperatorDebugCommand{Meta: Meta{Ui: ui}}
-
-	// Debug on the leader and all client nodes
-	code := cmd.Run([]string{"-address", url, "-duration", "250ms", "-server-id", "leader", "-node-id", "all", "-output", os.TempDir()})
-	assert.Equal(t, 0, code)
-
-	// Bad plugin name should be escaped before it reaches the sandbox test
-	require.NotContains(t, ui.ErrorWriter.String(), "file path escapes capture directory")
-	require.Contains(t, ui.OutputWriter.String(), "Starting debugger")
-
-	path := cmd.collectDir
-	defer os.Remove(path)
-
-	var pluginFiles []string
-	for _, pluginName := range cases {
-		pluginFile := fmt.Sprintf("csi-plugin-id-%s.json", helper.CleanFilename(pluginName, "_"))
-		pluginFile = filepath.Join(path, "nomad", "0000", pluginFile)
-		pluginFiles = append(pluginFiles, pluginFile)
-	}
-
-	testutil.WaitForFiles(t, pluginFiles)
-
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+type testCase struct {
+	name            string
+	args            []string
+	expectedCode    int
+	expectedOutputs []string
+	expectedError   string
 }
 
-func TestDebugUtils(t *testing.T) {
-	xs := argNodes("foo, bar")
-	require.Equal(t, []string{"foo", "bar"}, xs)
-
-	xs = argNodes("")
-	require.Len(t, xs, 0)
-	require.Empty(t, xs)
-
-	// address calculation honors CONSUL_HTTP_SSL
-	e := &external{addrVal: "http://127.0.0.1:8500", ssl: true}
-	require.Equal(t, "https://127.0.0.1:8500", e.addr("foo"))
+type testCases []testCase
 
-	e = &external{addrVal: "http://127.0.0.1:8500", ssl: false}
-	require.Equal(t, "http://127.0.0.1:8500", e.addr("foo"))
+func runTestCases(t *testing.T, cases testCases) {
+	t.Helper()
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			// Setup mock UI
+			ui := cli.NewMockUi()
+			cmd := &OperatorDebugCommand{Meta: Meta{Ui: ui}}
 
-	e = &external{addrVal: "127.0.0.1:8500", ssl: false}
-	require.Equal(t, "http://127.0.0.1:8500", e.addr("foo"))
+			// Run test case
+			code := cmd.Run(c.args)
+			out := ui.OutputWriter.String()
+			outerr := ui.ErrorWriter.String()
 
-	e = &external{addrVal: "127.0.0.1:8500", ssl: true}
-	require.Equal(t, "https://127.0.0.1:8500", e.addr("foo"))
+			// Verify case expectations
+			require.Equalf(t, code, c.expectedCode, "expected exit code %d, got: %d: %s", c.expectedCode, code, outerr)
+			for _, expectedOutput := range c.expectedOutputs {
+				require.Contains(t, out, expectedOutput, "expected output \"%s\", got \"%s\"", expectedOutput, out)
```

Comment:
The `%q` formatter does this for you:

```suggestion
				require.Contains(t, out, expectedOutput, "expected output %q, got %q", expectedOutput, out)
```

Reply:
Hadn't seen that -- much cleaner, thanks!